docs/builtins.md

```diff
@@ -604,3 +604,326 @@ num: 0
+
+#### Example
+
+```evy
+s := join ["a" "b" "c"] ", "
+print s
+```
+
+Output
+
+```evy:output
+a, b, c
+```
+
+#### Reference
+
+    join:string elems:[]string sep:string
+
+The `join` function takes two arguments: an array of strings and a
+separator string. The array of strings is the list of elements to be
+concatenated. The separator string is the string that will be placed
+between elements in the resulting string.
+
+The `join` function returns a single string that is the concatenation of
+the elements in the list of strings, with the separator string placed
+between elements.
+
+### `split`
+
+`split` splits a string into a list of substrings separated by the given
+separator. The separator can be any string, including the empty
+string.
+
+#### Example
+
+```evy
+print (split "a,b,c" ",")
+print (split "a,b,c" ".")
+print (split "a,b,c" "")
+```
+
+Output
+
+```evy:output
+[a b c]
+[a,b,c]
+[a , b , c]
+```
+
+#### Reference
+
+    split:[]string s:string sep:string
+
+The `split` function takes two arguments: the string to be split and the
+separator string. The string to be split is the string that will be
+split into substrings. The separator string is the string that will be
+used to split the string.
+
+The `split` function returns a list of substrings. The list of substrings
+contains all of the substrings of the original string that are
+separated by the separator string.
+
+If the string does not contain the separator, the `split` function returns
+an array of length 1 containing the original string.
+
+If the separator is the empty string, the `split` function splits the
+string after each character (UTF-8 sequence).
+
+If both the string and the separator are empty, the `split` function
+returns an empty list.
+
+### `upper`
+
+`upper` returns a string with all lowercase letters converted to
+uppercase.
+
+#### Example
+
+```evy
+s := upper "abc D e ü"
+print s
+```
+
+Output
+
+```evy:output
+ABC D E Ü
+```
+
+#### Reference
+
+    upper:string s:string
+
+The `upper` function takes a single argument: the string that should be
+converted to uppercase. The function returns a new string with all
+lowercase letters converted to uppercase.
+
+The `upper` function uses the Unicode character database to determine
+which characters are lowercase and which characters are uppercase.
+
+### `lower`
+
+`lower` returns a string with all uppercase letters converted to
+lowercase.
+
+#### Example
+
+```evy
+s := lower "abc D e ü"
+print s
+```
+
+Output
+
+```evy:output
+abc d e ü
+```
+
+#### Reference
+
+    lower:string s:string
+
+The `lower` function takes a single argument: the string that should be
+converted to lowercase. The function returns a new string with all
+uppercase letters converted to lowercase.
+
+The `lower` function uses the Unicode character database to determine
+which characters are lowercase and which characters are uppercase.
+
+### `index`
+
+`index` returns the index of a substring in a string, or -1 if the
```

Comment:
rather than `index` returns the index... how about `index` returns the position... just to give another word to help people who may not know exactly what "index" means in this context?

Reply:
yup, sounds good. done!